Feature: add libMdsFLib for old length arg

joshStillerman opened the pull request:
A new version of MdsLib especially for broken old fortran programs
that does not try to overwrite length argument if it is 1 by
reference.

joshStillerman committed:
Feature: add libMdsFLib for old length arg

A new version of MdsLib especially for broken old fortran programs
that does not try to overwrite length argument if it is 1 by
reference.

joshStillerman committed:
Feature: new library MdsFLib with old size arg

joshStillerman commented:
retest this please

joshStillerman committed:
temporary printf to see WTF is going on

joshStillerman committed:
use calloc to insure buffer is clear

joshStillerman committed:
calloc call wrong

joshStillerman committed:
Can not figure out why it appears to be using MdsFLib, or MdsLib has the if FORTRAN

joshStillerman committed:
typo

joshStillerman committed:
fix windoz link (I hope)

joshStillerman commented:
restest this please

joshStillerman committed:
typos - thanks @stephen

joshStillerman commented:
retest this please

joshStillerman commented:
retest this please

WhoBrokeTheBuild commented on mdslib/Makefile.in:
```diff
@@ -62,8 +69,20 @@ $(MdsLib): $(OBJECTS)
+MdsLib_client.o: MdsLib.c
 	$(COMPILE.c) $< $(OUTPUT_OPTION) -DFILENAME=MdsLib.c -D_CLIENT_ONLY
+
+MdsFLib.o: MdsLib.c
```
Maybe we should make it `MdsLib_fortran` to match the `client` one above

WhoBrokeTheBuild commented on mdslib/testing/mdslib_ftest.f:
```diff
@@ -27,9 +27,9 @@ program mdslib_test
       enddo
 
       dsc = descr(IDTYPE_FLOAT,result,0,0)
```
I think we should add another test at the bottom, instead of replacing the last test

WhoBrokeTheBuild replied:
@joshStillerman I would still put this test back as it was, and add another test that tests the `size=1`

joshStillerman commented:
2 good suggestions.  Though

  - if Andrea is correct we need to this all in Makefile.am

  - It will not solve the memtest

  - Tom Suggests:

> sigaction(SIGSEGV, SIG_IGN, 0);
> *length=value
> sigaction(SIGSEGV,SIG_DFL, 0);
>
> or you could use the "old action" argument to get the old handler and 
> put it back.
Thoughts ?

On 3/30/21 2:51 PM, Stephen Lane-Walsh wrote:
>
> *@WhoBrokeTheBuild* requested changes on this pull request.
>
> ------------------------------------------------------------------------
>
> In mdslib/Makefile.in 
> <https://github.com/MDSplus/mdsplus/pull/2264#discussion_r604346326>:
>
> >  	$(COMPILE.c) $< $(OUTPUT_OPTION) -DFILENAME=MdsLib.c -D_CLIENT_ONLY
> +
> +MdsFLib.o: MdsLib.c
>
> Maybe we should make it |MdsLib_fortran| to match the |client| one above
>
> ------------------------------------------------------------------------
>
> In mdslib/testing/mdslib_ftest.f 
> <https://github.com/MDSplus/mdsplus/pull/2264#discussion_r604351251>:
>
> > @@ -27,9 +27,9 @@ program mdslib_test
>         enddo
>   
>         dsc = descr(IDTYPE_FLOAT,result,0,0)
>
> I think we should add another test at the bottom, instead of replacing 
> the last test
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/MDSplus/mdsplus/pull/2264#pullrequestreview-624592009>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABY5AZJ3PNQYOL6OA46437LTGIMS7ANCNFSM4ZYDFSNQ>.
>
-- 
*Joshua Stillerman*
Research Engineer
MIT Plasma Science and Fusion Center
617.253.8176
***@***.*** ***@***.***>

WhoBrokeTheBuild commented:
There is no Makefile.am for Mdslib, there is one for the Mdslib/testing however. All of this looks correct to me.

WhoBrokeTheBuild commented on mdslib/testing/Makefile.am:
```diff
@@ -5,7 +5,7 @@ include ../../testing/testing.am
 
 AM_CFLAGS = $(TARGET_ARCH) $(WARNFLAGS) $(TEST_CFLAGS)
 AM_LDFLAGS = -L@MAKESHLIBDIR@ $(RPATHLINK),@MAKESHLIBDIR@ @LIBS@
-LDADD = -lMdsLib -lTdiShr -lTreeShr -lMdsShr -lMdsIpShr $(TEST_LIBS)
```
We should continue testing MdsLib and add a separate test for MdsLib_fortran later

joshStillerman committed:
Fix: rename MdsFLib - MdsLib_fortran

change the name to correspond to _client name.
remove the test of FLib (failing - to be dealt with later)

joshStillerman committed:
fix Makefile.in
